### PR TITLE
Improve gir file lookup

### PR DESCRIPTION
**Pull request by JustABanana**
This makes the getGirDirectory function return multiple possible
candidate directories and makes it behave according to the XDG_DATA_DIRS
spec on linux.

Resolves https://github.com/gtkd-developers/gir-to-d/issues/36

Please note that:
1. I haven't tested this commit on systems other than linux
2. I'm just learning D/gtk

### CODE REVIEW

File: source/gtd/GirWrapper.d [hunk 1]
```diff
@@ -585,56 +588,56 @@ class GirWrapper
...
+		if(dirs !is null)
 		{
-			import std.process : environment;
-
-			static string path;
+			return dirs;
+		}
+		else
+		{
+			dirs = [];
```
Comment (MikeWey) [inline source/gtd/GirWrapper.d:600]: This isn't necessary in D, you can just append to an empty array and the runtime will allocate it for you.
I would remove the whole else clause in this case.

File: source/gtd/GirWrapper.d [hunk 2]
```diff
@@ -585,56 +588,56 @@ class GirWrapper
...
+			dirs ~= commandlineGirPath;
+		}
 
+		version(Windows)
+		{
 			foreach (p; splitter(environment.get("PATH"), ';'))
 			{
 				string dllPath = buildNormalizedPath(p, "libgtk-3-0.dll");
 
-				if ( exists(dllPath) )
```
Comment (MikeWey) [inline source/gtd/GirWrapper.d:603]: The check to see if the path to de dll's exists should stay, or else you will create a lot of bogus paths.

File: source/gtd/GirWrapper.d [hunk 3]
```diff
@@ -585,56 +588,56 @@ class GirWrapper
...
-				path = path.buildNormalizedPath("../share/gir-1.0");
+				dirs ~= path.buildNormalizedPath("../share/gir-1.0");
 			}
-			else
-			{
-				path = environment.get("HOMEBREW_ROOT");
-				if(path)
+			path = environment.get("HOMEBREW_ROOT");
+			if(path)
 				{
```
Comment (MikeWey) [inline source/gtd/GirWrapper.d:627]: The brackets and contents of the if indented to far.

File: source/gtd/GirWrapper.d [hunk 4]
```diff
@@ -585,56 +588,56 @@ class GirWrapper
...
-
-			return path;
 		}
 		else
 		{
-			return "/usr/share/gir-1.0";
+			string xdgDataDirs = environment.get("XDG_DATA_DIRS", "/usr/local/share:/usr/share");
+		        foreach (p; splitter(xdgDataDirs, ':'))
+		        {
+		        	dirs ~= p.buildNormalizedPath("gir-1.0");
```
Comment (MikeWey) [inline source/gtd/GirWrapper.d:636]: Should we check if the path exists before appending it to the result?

**Review by MikeWey — CHANGES_REQUESTED** [review]
Looks good apart from a few minor details.

File: source/gtd/GirWrapper.d (hunk 4 above)
Comment (JustABanana) [reply source/gtd/GirWrapper.d:636]: I could also replace this with a filter before the return that removes the directories that don't exist. I wanted to have the caller responsible for checking if the dirs actually exist, but adding a check to make sure the path actually exists may be a good idea

File: source/gtd/GirWrapper.d (hunk 2 above)
Comment (JustABanana) [reply source/gtd/GirWrapper.d:603]: I think this was fixed by https://github.com/gtkd-developers/gir-to-d/pull/37/commits/df3895d13030d7a71dce5a97db37e2afcba0fd64

**Review by MikeWey — APPROVED** [review]

**Comment by MikeWey** [comment]
Thanks.

**Pull request merged**